### ISSUE TITLE
Add interaction between the Schedule tab and Work tab.

### ISSUE THREAD

**Issue by basicsmods** [fluffy-mods/WorkTab#211]
What I'm mainly curious about is whether you think this is reasonably feasible. If you think it is, I'd like to try working on it myself. If you don't think it's possible or you think it would be a massive undertaking then I'll just forget about it.

The idea is basically to give your scheduled "jobs" (so for example a 4 hour time range in the Schedule tab being set to sleep/recreation/meditation) a priority that interacts with the work tab priorities. 

Let's say that you want to prioritize growing anima grass so you want your pawns to be meditating as much as possible. You want one of your pawns to be set to Meditate for 18h per day (with the other 6h set to sleep) so you set that in the Schedule tab. But that pawn is also your cook. So you want them to go cook whenever there's cooking jobs and to meditate / sleep otherwise. Using this hypothetical feature, you set the "Schedule priority" to 2 and you set that pawn's Cooking priority to 1.

Another example is with doctoring. It's very annoying to have someone bleeding out in the bed 2 feet away from your sleeping / meditating doctor and your doctor's like "damn that pawn could really use a doctor, but it's bed / meditation time for me rip". This feature would fix this issue (as long as the doctor was actually sleeping / meditating due to the Schedule tab and not because of their basic needs). (I know there are other mods that try to specifically fix this "lazy doctoring" issue, but a) they don't seem to work all that well, and b) this is just an example for a feature that would be more generally useful than just this example.)

More generally, you give the Schedule tab itself a priority. Anytime a pawn is doing something _specifically because the schedule tab is "telling" them to do it_ they will ignore the schedule tabs wishes if there is an available job with higher priority. 

Pawns still do some of those things (meditate, recreation, sleep) even when they are not scheduled. The pawn reached a threshold and decided they need to do those things. Those times should be unaffected by this feature. This feature should only "commandeer" the pawn's current task when they are doing something specifically due to a schedule tab node.

This could potentially be tuned to be a bit smarter (and maybe more efficient from a CPU perspective) by only checking once every half hour (or any arbitrary time length that seems appropriate) so the pawn isn't constantly running back and forth between their bed / meditation spot and a "job-site" that keeps getting a new task available. 

So anyways, do you think this is reasonably achievable? 

PS. Thanks for the incredible mod and all the other mods you've made and maintain. It cannot be overstated how huge of an impact your work (and the work of other large scale modders) have on this game. The community is in debt to you all.

**Comment by basicsmods** [comment]
Just from reading the description of one of the specific "lazy doctor" mods, it sounds like the game has some tasks designated as "emergency" which (**I'm just speculating here**) means the pawn will abandon their needs / schedule in favour of that task when it's available. For example, firefighting I believe is set as an emergency task by default and so your pawns will abandon "need" tasks if there's a fire in the Home area. 

If this is truly how that system works, I wonder if there'd be a simple way to try to achieve something similar to this feature request where we just say any task set to P1 will be given that "emergency" designation. (Would definitely want this to be optional so either coming as a separate mod or a setting in this mod that defaults to Off.)

Depending on how this "emergency" system works, it'd likely be a bit different than what I'm describing (my feature request involves _not_ bailing out of tasks that the pawn is doing due to their needs and only overriding when the pawn is doing something specifically because the Schedule tab is telling them to). But it might achieve something very similar with a lot less effort so it could be worth looking into.

**Comment by Doomster14** [comment]
Good idea. It needs much thinking though. I see a good solution more like an overhaul of logic behind Schedule and Work tabs. Currently, Schedule is the boss. So, if a pawn decides to recreate, nothing else matters. It feels like Recreation is an unseen first column in Work tab with priority of 1.

I don't plan to mess with this, it seems complicated: https://rimworldwiki.com/wiki/Menus#Schedule
